### PR TITLE
fix #18113

### LINKED ISSUE #18113: Sequtils bug: cannot attach a custom pragma to 's2'
A combination of pragma + `toSeq` + `filterIt` causes this.

### Example
```nim
import sequtils

{.push raises: [Defect].}

var a = toSeq([1, 2, 10, 20]).filterIt(it > 5)
```

### Current Output
```
sequtils.nim(738, 18) Error: cannot attach a custom pragma to 's2'
```

### Possible Solution

Several walkarounds:

```nim
import sequtils

# {.push raises: [Defect].}         <--- commenting out this line

var a = toSeq([1, 2, 10, 20]).filterIt(it > 5)
```

```nim
import sequtils

{.push raises: [Defect].}

var a = toSeq([1, 2, 10, 20])     # <--- doing toSeq and filterIt separately
var b = a.filterIt(it > 5)
```

### Additional Information

This is a regression, and `git bisects` says it is caused by this PR: https://github.com/nim-lang/Nim/pull/17448. cc @saem 

```
$ nim -v
Nim Compiler Version 1.5.1 [Linux: amd64]
Compiled at 2021-05-27
Copyright (c) 2006-2021 by Andreas Rumpf

git hash: 1e0165186bb8539cfd8aca1a7af8d01dc278bd46
active boot switches: -d:release
```

### PR DESCRIPTION
**Pull request by saem**
fix #18113

### CODE REVIEW

**Comment by saem** [comment]
@narimiran have a look

File: tests/proc/t17157.nim [hunk 1]
```diff
@@ -1,5 +1,6 @@
 discard """
   errormsg: "'untyped' is only allowed in templates and macros or magic procs"
+  disabled: true
```
Comment (timotheecour) [inline tests/proc/t17157.nim:3]: @saem  why was this disabled? is that a regression?

File: tests/template/t18113.nim [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# ensure template pragma handling doesn't eagerly attempt to add an implicit
+# 'pushed' pragma to the evaluation of any intermediate AST prior to
+# substitution.
+ 
+# bug #18113
+
+import sequtils
+
+{.push raises: [Defect].}
```
Comment (timotheecour) [inline tests/template/t18113.nim:9]: `raises: [Defect]` doesn't make a lot of sense even if it was in #18113; instead this tests should use a push pragma that makes sense and reproduced the bug, eg:
```nim
{.push inline.}
var a = toSeq([1, 2, 3, 5, 10]).filterIt(it > 5)
{.pop.}
```

File: tests/proc/t17157.nim (hunk 1 above)
Comment (narimiran) [reply tests/proc/t17157.nim:3]: It was me who disabled it (after the consultation with @Araq), not @saem.

And the reason is that it was more important to have this fix merged than this test green.

**Review by ttloves2 — COMMENTED** [review]
?

**Pull request merged**